### ISSUE TITLE
zlib lacks zlib-config.cmake/zlibconfig.cmake which makes  folly package unusable with x64-static using CMAKE

### ISSUE THREAD

**Issue by jz-fb** [microsoft/vcpkg#13287]
###  In cmakelists.txt:
`set(CMAKE_TOOLCHAIN_FILE "../thirdparty/vcpkg/scripts/buildsystems/vcpkg.cmake") ` **<-- toolchain added**
`set(folly_DIR "${CMAKE_SOURCE_DIR}/../thirdparty/vcpkg/packages/folly_x64-windows-static/share/folly") `**<-- point to follyconfig.cmake for find_package() to use**

##  Problem:
ZLIB package lacks a zlibconfig.cmake (also uppercase v lowercase may also be an issue). Trying to build without folly_DIR also fails with find_package()  being unable to find folly.

##  Error Msg:
-- Found folly: C:/Users/pcc/Desktop/disp2/thirdparty/vcpkg/packages/folly_x64-windows-static
-- Configuring done
CMake Error at benchmarks/CMakeLists.txt:76 (add_executable):
  Target "future_benchmark" links to target "ZLIB::ZLIB" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

CMake Error at benchmarks/CMakeLists.txt:76 (add_executable):
  Target "future_benchmark" links to target "ZLIB::ZLIB" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

CMake Error at benchmarks/CMakeLists.txt:76 (add_executable):
  Target "future_benchmark" links to target "ZLIB::ZLIB" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

CMake Error at benchmarks/CMakeLists.txt:76 (add_executable):
  Target "future_benchmark" links to target "ZLIB::ZLIB" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.

**Comment by Neumann-A** [comment]
title has it completely wrong. `FindZLIB.cmake` is supplied by CMake

seems like target `future_benchmark` is simply missing a `find_package(ZLIB)/find_dependency(ZLIB)` call before defining the targets.

>-- Found folly: C:/Users/pcc/Desktop/disp2/thirdparty/vcpkg/packages/folly_x64-windows-static
Trying to build without folly_DIR

also looks suspicious. 


(Note: looked into the portfile and seems like folly's portfile is doing some shenanigans with the `-config.cmake` which need a bit more corrections)

**Comment by Neumann-A** [comment]
also duplicate of #13196
closed by #13278

**Comment by NancyLi1013** [comment]
Hi @jz-fb 
Thanks for posting this issue.
Please try to rebuild this based on #13278.

**Comment by jz-fb** [comment]
Just tried that branch, but I'm still getting the zlib issues. I now am able to use `find_package()` rather than `set(folly_DIR "${CMAKE_SOURCE_DIR}/../thirdparty/vcpkg/packages/folly_x64-windows-static/share/folly"),` but  can't do much with the zlib issue.  Some things to note: 1) I deleted the prior vcpkg folder before cloning and making sure I have that branch checked out. 2) I wiped the build folder before doing `cmake ..`. 3) nowhere in my cmakelists.txt do I make reference to zlib and only happens when I enable sections of the cmakelists that deal with folly.
